app/assets/javascripts/components/buttons/generic-button-component.js

```diff
@@ -0,0 +1,41 @@
+ManageIQ.angular.app.component('genericButtonComponent', {
+  bindings: {
+    angularForm: '=',
+    modelCopy: '=',
+    model: '=',
+    redirectUrl: '=?',
+    newRecord: '<',
+    entity: '=?',
+    entityName: '=?',
+    onSaveClicked: '&?',
+    onAddClicked: '&?',
+  },
+  controllerAs: 'vm',
+  controller: genericButtonController,
+  templateUrl: '/static/buttons/generic_button_component.html.haml',
+});
+
+genericButtonController.$inject = ['miqService', '$window'];
+
+function genericButtonController(miqService) {
+  var vm = this;
+
+  vm.$onInit = function() {
+    vm.saveable = miqService.saveable;
+
+    vm.resetClicked = function() {
```

Comment:
> Or have a miqService function that injects the default implementations of those funcions on the controller object.

`miqService` should be the last resort not the first choice.
Otherwise it will end up being like `application_controller`.

> We can always put the common code in a FormController and inherit all the *FormControllers from that one

This approach seems interesting though.